RpcSystemAssetTracker/UserUnspentCoinOutputsKey.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Neo.IO;
+
+namespace Neo.Plugins
+{
+    public class UserUnspentCoinOutputsKey : IComparable<UserUnspentCoinOutputsKey>, IEquatable<UserUnspentCoinOutputsKey>,
+        ISerializable
+    {
+        public bool IsGoverningToken; // It's either the governing token or the utility token
+        public readonly UInt160 UserAddress;
+        public readonly UInt256 TxHash;
+
+        public int Size => 1 + UserAddress.Size + TxHash.Size;
+
+        public bool Equals(UserUnspentCoinOutputsKey other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return IsGoverningToken == other.IsGoverningToken && Equals(UserAddress, other.UserAddress) && Equals(TxHash, other.TxHash);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != this.GetType()) return false;
+            return Equals((UserUnspentCoinOutputsKey) obj);
+        }
+
+        public int CompareTo(UserUnspentCoinOutputsKey other)
+        {
+            if (ReferenceEquals(this, other)) return 0;
```

Comment:
Are these checks here in CompareTo correct?
In the other cases of equal it returns false when compared to null.

In this sense, compare other to null and turn 1 means that a not null object is better, right?

Reply:
Returning 1 from compareTo means the objects are not equal, and that a non-null object is greater. I believe this implementation is correct; however, it should not be called with a null other object in this code anyway. It is defensive.